modules/openapi-generator/src/test/java/org/openapitools/codegen/kotlin/KotlinClientCodegenModelTest.java

```diff
@@ -322,9 +322,9 @@ public void complexPropertyTest() {
     public static Object[][] modelNames() {
         return new Object[][]{
                 {"TestNs.TestClass", new ModelNameTest("TestNs.TestClass", "TestNsTestClass")},
-                {"$", new ModelNameTest("$", "Dollar")},
+                {"$", new ModelNameTest("$", "Value")},
                 {"for", new ModelNameTest("`for`", "For")},
-                {"One<Two", new ModelNameTest("One<Two", "OneLessThanTwo")},
+                {"One<Two", new ModelNameTest("One<Two", "OneTwo")},
```

Comment:
Looks to me `Dollar` and `OneLessThanTwo` should be the correct values

Reply:
Hi @wing328 -- thanks for the feedback! Can you elaborate a little more on why you would expect `Dollar` and `OneLessThanTwo` to be the expected values? I'm sure I'm missing something, but calling out those two in particular seems somewhat arbitrary to me at the moment.

Reply:
@wing328 -- just wanted to check in and see if there was any particular reason for `$` and `<` to be converted in class names. I'm happy to change that, was just hoping to learn why the current behavior is expected specifically just for those two characters. Thank you!

Reply:
Hi @wing328 -- I'd like to get this merged, and I'm not attached to these particular use cases you've called out (I'm only really concerned about `-` being sanitized as `Minus`). If I make this change are you comfortable approving/merging this PR?

Reply:
@tmeneau thanks again for the PR. I would prefer to keep the current behaviour (replacing symbols with the english names) and removing this from this PR will likely introduce breaking changes to the existing users of the generator.

cc @Zomzog - the author of https://github.com/OpenAPITools/openapi-generator/pull/2430

Reply:
@wing328  / @Zomzog 

I'm really interested in the outcome of this discussion. For me, it was highly unexpected that when using a Kotlin generator I get a class named `SomeMinusClassMinusName` and when using `spring` I get my expected `SomeClassName`.

For our current project, it would be a reason to step away from generating Kotlin code which would be a real pity but our schema is provided to us.

Reply:
@wing328 @Zomzog I've updated this PR to only change the sanitization of "minus". I'd think that ideally the Kotlin generator would follow the Java conventions, but the "minus" sanitization is the sanitization that's truly affecting us (and apparently @martinvw).

Let me know if this is not acceptable to you, or if you have any additional concerns about this. Otherwise I'm hoping we can get this merged in so we can start using the kotlin generator.

Thank you for your time reviewing and discussing this PR, and for all the work you do on this project!

Reply:
Another option here would be to add a configuration flag, something along the lines of, `sanitizeModelWithJavaConventions`. The default behavior would remain the same for any users that rely on the Kotlin sanitization, but users such as myself and @martinvw could opt into sanitization identical to the Java generation with a configuration flag.

@wing328 @Zomzog would you feel more comfortable with that approach?